appcli/orchestrators.py

```diff
@@ -598,6 +599,239 @@ def __exec_command(self, command: Iterable[str]) -> CompletedProcess:
         return subprocess.run(command, capture_output=False)
 
 
+class HelmOrchestrator(Orchestrator):
+    """
+    Uses helm to provision a kubernetes backed helm chart.
+    """
+
+    def __init__(
+        self,
+        release_name: (str),
+        chart_location: Path = Path("chart"),
+        dev_chart_location: Optional[Path] = None,
+    ):
+        """
+        Creates a new instance of an orchestrator for helm-based applications.
+
+        NOTE: All `Path` objects are relative to the configuration directory.
+
+        Args:
+            release_name (str): A name to give to this helm release.
+            chart_location (Path): Path to the helm chart file/directory to deploy.
+            dev_chart_location (Path | None):
+                [Optional] An absolute path to the chart source for when deploying in `dev_mode`.
+        """
+        self.release_name = release_name
+        self.chart_location = chart_location
+        self.dev_chart_location = dev_chart_location
+
+    def start(
+        self, cli_context: CliContext, service_name: tuple[str, ...] = None
+    ) -> CompletedProcess:
+        """
+        Installs (or upgrades) a helm chart inside the current kubernetes cluster.
+
+        Args:
+            cli_context (CliContext): The current CLI context.
+
+        Returns:
+            CompletedProcess: Result of the orchestrator command.
+        """
+        # Generate the command string.
+        command = [
+            "helm",
+            "upgrade",
+            "--install",
+            "--namespace",
+            self.release_name,
+            "--create-namespace",
+        ]
+        # Set values args.
+        for arg in self.__generate_values_args(
+            helm_values_dir=cli_context.get_helm_values_dir(),
+            helm_values_files_dir=cli_context.get_helm_values_files_dir(),
```

Comment:
Another benefit of this being instance vars in the class, is that we can calculate the list of args in the contructor.

Reply:
We can't do it in the constructor as it doesn't get access to `cli_context.get_generated_configuration_dir()` which we need to find the absolute path to the files.